torch/distributed/_tensor/debug/comm_mode.py

```diff
@@ -3,6 +3,11 @@
 
 import torch
 from torch.distributed._tensor.api import DTensor
+
+from torch.distributed._tensor.examples.advanced_module_tracker import (
```

Comment:
IMO it's not ideal to import things from `examples` here.

Reply:
Maybe move this Module in common_dtensor.py since we are re-using it a couple places?
https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/distributed/_tensor/common_dtensor.py

Reply:
Yeah I agree, let's move it into `comm_mode.py` since users always use this feature through comm_mode.